vmdb/app/models/mixins/ems_openstack_mixin.rb

```diff
@@ -27,7 +27,9 @@ def openstack_handle(options = {})
       raise "no credentials defined" if self.missing_credentials?(options[:auth_type])
 
       username = options[:user] || self.authentication_userid(options[:auth_type])
-      password = options[:pass] || self.authentication_password(options[:auth_type])
+
+      # the |"" is here to prevent an unhelpful ArgumentError from OpenStackHandle::Handle#connect
+      password = options[:pass] || self.authentication_password(options[:auth_type]) || ""
```

Comment:
The only thing I'd change is:

```ruby
def required_credential_fields(type)
  case type
  when :type_a then [:userid, :password]
  when :type_b then [:userid]
end
```

Something like that...  Allow the including class to customize the list of required credentials for each authentication type.

Other than that, :+1: